File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/UserCreateDTO.java
package com.szlazakm.chatserver.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Value;
import lombok.experimental.SuperBuilder;
import lombok.extern.jackson.Jacksonized;

@Value
@SuperBuilder
@Jacksonized
public class UserCreateDTO {

    @NotBlank
    @Pattern(regexp = "^[a-zA-Z]")
    String firstName;
    String lastName;
    String identityKey;
    String pin;
    String phoneNumber;

}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/UserCreateDTO.kt
package com.szlazakm.safechat.webclient.dtos

import com.fasterxml.jackson.annotation.JsonProperty

data class UserCreateDTO(
    @JsonProperty("firstName")
    val firstName: String,
    @JsonProperty("lastName")
    val lastName: String,
    @JsonProperty("phoneNumber")
    val phoneNumber: String,
    @JsonProperty("identityKey")
    val identityKey: String,
    @JsonProperty("pin")
    val pin: String
)